sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuParquetFileFormat.scala

```diff
@@ -97,18 +97,54 @@ object GpuParquetFileFormat {
     }
   }
 
-  def getPrecisionList(schema: StructType): Seq[Int] =  {
-    def precisionsList(t: DataType): Seq[Int] = {
-      t match {
-        case d: DecimalType => List(d.precision)
-        case _: StructType =>
-          throw new IllegalStateException("structs are not supported right now")
-        case _: ArrayType =>
-          throw new IllegalStateException("arrays are not supported right now")
-        case _ => List(0)
+  def parquetWriterOptionsHelper(schema: StructType): ParquetWriterOptions.Builder[_] = {
```

Comment:
This could use a better name and some documentation on what it does.  "Helper" doesn't convey much about what it's doing.  How about something more descriptive, like `writerOptionsFromSchema`?